psi4/driver/procrouting/dft_funcs/dict_builder.py

```diff
@@ -0,0 +1,242 @@
+    else:
+        aliases = [functional_dict["name"].upper()]
+    return aliases
+
+# alias table for dispersion
+dispersion_names = {}
+dispersion_names["d2p4"] = "d2p4"
+dispersion_names["d2gr"] = "d2gr"
+dispersion_names["d3zero"] = "d3zero"
+dispersion_names["d3bj"] = "d3bj"
+dispersion_names["d3mzero"] = "d3mzero"
+dispersion_names["d3mbj"] = "d3mbj"
+dispersion_names["d"] = "d2p4"
+dispersion_names["d2"] = "d2p4"
+dispersion_names["d3"] = "d3zero"
+dispersion_names["d3m"] = "d3mzero"
+
+functionals = {}
+for functional_name in dict_functionals.keys():
+    functional_aliases = get_functional_aliases(dict_functionals[functional_name])
+    # first process aliases of parent functional
+    for alias in functional_aliases:
+        functionals[alias] = dict_functionals[functional_name]
+    # if the functional is already dispersion corrected, skip to next
+    if "dispersion" in dict_functionals[functional_name].keys():
+        continue
+    # loop through dispersion types in dashparams incl aliases 
+    # and build dispersion corrected version incl aliases
+    for dispersion_name in dispersion_names.keys():
+        dispersion_type = dispersion_names[dispersion_name]
+        for dispersion_functional in dashcoeff[dispersion_type].keys():
+            if dispersion_functional.upper() in functional_aliases:
+                func = copy.deepcopy(dict_functionals[functional_name])
+                func["name"] = func["name"] + "-" + dispersion_type
+                func["dispersion"] = dict()
+                if "citation" in dashcoeff[dispersion_type][dispersion_functional].keys():
+                    func["dispersion"]["citation"] = dashcoeff[dispersion_type][dispersion_functional].pop("citation")
+                func["dispersion"]["type"] = dispersion_type
+                func["dispersion"]["params"] = dashcoeff[dispersion_type][dispersion_functional]
+                for alias in functional_aliases:
+                    alias = alias + "-" + dispersion_name.upper()
+                    functionals[alias] = func
+
+def check_consistency(func_dictionary, func_name):
+    """
+    This checks the consistency of the definitions of exchange and correlation components
+    of the functional, including detecting duplicate requests for LibXC params, inconsistent
+    requests for HF exchange and missing correlation.
+    """
+    # 1a) sanity checks definition of xc_functionals
+    if "xc_functionals" in func_dictionary.keys():
+        if "x_functionals" in func_dictionary.keys() or "x_hf" in func_dictionary.keys():
+            raise ValidationError("SCF: Duplicate specification of exchange (XC + X) in functional %s." % (name))
+        elif "c_functionals" in func_dictionary.keys() or "c_mp2" in func_dictionary.keys():
+            raise ValidationError("SCF: Duplicate specification of correlation (XC + C) in functional %s." % (name))
+    # 1b) require at least an empty exchange functional entry or X_HF
```

Comment:
Can you add a space above a comment/code section to break up this code a bit for readability?

Reply:
Fixed. I've added another docstring further below.